bitcoin/src/consensus/encode.rs

```diff
@@ -499,9 +502,7 @@ impl Encodable for String {
     #[inline]
     fn consensus_encode<W: io::Write + ?Sized>(&self, w: &mut W) -> Result<usize, io::Error> {
         let b = self.as_bytes();
-        let vi_len = VarInt(b.len() as u64).consensus_encode(w)?;
-        w.emit_slice(b)?;
-        Ok(vi_len + b.len())
+        Ok(w.emit_varint(b.len())? + w.emit_slice(b)?)
```

Comment:
This is cute but does it really make sense to complicate `emit_slice` by returning `usize` just because of this? This also repeats so why not have `fn emit_varint_prefixed_slice`?

Reply:
That actually exists as

Reply:
Changed the two other occurrences to use this method.

I'd be ok with changing that, but our `Encodable` trait also does it and now `emit_varint` has to do it, so it's not the only remaining method breaking with the trait. Tbh the int ones can also do it to make all the `len += ` pieces more concise, but I didn't do it because it would probably be a ton of changes. I can do it if you prefer.

Reply:
On second thought, I wouldn't be opposed to moving this `consensus_encode_with_size` method into the `WriteExt` trade as something like `emit_slice_with_length` like you said.

Reply:
Yeah, moving it here sounds good. But my point is we shouldn't return `usize` from `emit_slice` - the length of the slice is already known. Also I wonder if we really want it rather than just using `write_all`.